perftests/network-ssl/main.c

```diff
@@ -0,0 +1,133 @@
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include "asprintf.h"
+#include "events.h"
+#include "sock.h"
+#include "warnp.h"
+
+#include "network_ssl.h"
+
+static int conndone = 0;
+
+static int
+callback_write_test(void * cookie, ssize_t val)
+{
+
+	(void)cookie;
+
+	/* We should have written a single byte, or received an error. */
+	assert((val == 1) || (val == -1));
+
+	/* An error occurred. */
+	if (val == -1)
+		return (-1);
+
+	/* We can stop the event loop peacefully. */
+	conndone = 1;
+
+	/* Success! */
+	return (0);
+}
+
+static int
+do_request(struct network_ssl_ctx * ctx)
+{
+	const uint8_t req = 0;
+
+	/* Send a 1-byte buffer containing 0. */
+	if (network_ssl_write(ctx, &req, 1, 1, callback_write_test, NULL)
```

Comment:
Can't pass a pointer here which will go out of scope before the operation completes.  Did you want to declare `req` as `static`?

Reply:
eh?  I've got the `events_spin()` inside this function, so I think it's valid?

Admittedly, you seem to prefer `events_spin()` inside `main()`, so I should probably follow that pattern (even though I'm personally a fan of smaller `main()`.

Reply:
Oh.  Err, yeah, I didn't read far enough and just assumed that the event loop would be in `main`.  Doesn't really matter where you put it in this case but in the past I've kept the spinning in `main` so that I could test launching a bunch of requests at once.